### PR TITLE
Publish new article

### PR DESCRIPTION
**Pull request by Erikaguilera123**
## Sumary
# I made the automated test to publish the new article and created the method "Login" in the class "Helper"

### CODE REVIEW

File: cypress/const/helpers.js [hunk 1]
```diff
@@ -1,7 +1,20 @@
...
 
 export function RANDOM (){
   Math.floor(Math.random() * 100) + 1;
 }
+
+export function LOGIN (){
+  cy.get(IDs.idSingIn).click();
+
+  HAVETEXT(IDs.email,VAR.emailRegister);
+  HAVETEXT(IDs.password,VAR.passRegister);
```
Comment (betoobernal) [inline cypress/const/helpers.js]: add spaces
```suggestion
  HAVETEXT(IDs.email, VAR.emailRegister);
  HAVETEXT(IDs.password, VAR.passRegister);
```

File: cypress/integration/loginAngular.spec.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+  it('Sign up validate null', () => { 
+    cy.get(IDs.btnSingIn).should('not.have.class', 'disabled');
+
+   });
+
+  it('Sign up data incorrect', () => { 
+    
+    HAVETEXT(IDs.email,VAR.textSmall);
+    HAVETEXT(IDs.password,VAR.textSmall);
```
Comment (betoobernal) [inline cypress/integration/loginAngular.spec.js]: add spaces
```suggestion
    HAVETEXT(IDs.email, VAR.textSmall);
    HAVETEXT(IDs.password, VAR.textSmall);
```

File: cypress/const/constants.js [hunk 3]
```diff
@@ -7,9 +7,19 @@ export const VAR = {
...
+  userInvalid: "username can't be blank",
+  emailRegister: 'correoprueba2@correo.co',
+  passRegister: '1234567890',
+  error: 'email or password is invalid',
+  articuleTitleInvlid: "title can't be blank,is too short (minimum is 1 character)",
+  articuleBodyInvalid: "body can't be blank",
+  articuleDescripInvalid: "description can't be blank,is too short (minimum is 1 character)",
+  title: "Titulo",
+  descripcion: "textotextotextotextotextotexto ",
+  body: "bodybodybodybodybodybodybodybodybody "
```
Comment (eledwinn) [inline cypress/const/constants.js]: Use single quotes whenever possible. Please check if your vscode have tslint installed.

File: cypress/integration/newArticule.spec.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import { VAR, URL } from '../const/constants';
+import { IDs } from '../page-objects/newArticule';
+import{ HAVETEXT, LOGIN } from '../const/helpers';
+
+context('New articule', () => {
+  beforeEach(() => {
+    cy.visit(URL.index);
+    LOGIN ();
```
Comment (betoobernal) [inline cypress/integration/newArticule.spec.js]: remove space
```suggestion
    LOGIN();
```

File: cypress/integration/newArticule.spec.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+     cy.get(`.${IDs.error}`).contains(VAR.articuleTitleInvlid)
+     cy.get(`.${IDs.error}`).contains(VAR.articuleBodyInvalid)
+     cy.get(`.${IDs.error}`).contains(VAR.articuleDescripInvalid);
+    });
+
+    it('Public new articule', () => { 
+
+      HAVETEXT(IDs.title,VAR.title);
+      HAVETEXT(IDs.descripcion,VAR.descripcion);
+      HAVETEXT(IDs.body,VAR.body)
```
Comment (betoobernal) [inline cypress/integration/newArticule.spec.js]: add spaces
```suggestion
      HAVETEXT(IDs.title, VAR.title);
      HAVETEXT(IDs.descripcion, VAR.descripcion);
      HAVETEXT(IDs.body, VAR.body)
```

File: cypress/const/helpers.js [hunk 6]
```diff
@@ -1,7 +1,20 @@
+import { VAR } from '../const/constants';
+import { IDs } from '../page-objects/loginAngular';
+
 export function HAVETEXT (elemnt, text) {
```
Comment (eledwinn) [inline cypress/const/helpers.js]: Please use cammelCase for function names. Example: `haveText` and `random`

File: cypress/integration/loginAngular.spec.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    cy.get(IDs.btnSingIn).click();
+
+    cy.get(`.${IDs.errorRegister}`).contains(VAR.error);
+
```
Comment (eledwinn) [inline cypress/integration/loginAngular.spec.js]: Remove empty line

File: cypress/integration/newArticule.spec.js (hunk 4 above)
Comment (eledwinn) [inline cypress/integration/newArticule.spec.js]: cammelCase here, and no white space between name and parenthesis.

File: cypress/integration/newArticule.spec.js [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+    LOGIN ();
+    cy.get(IDs.newArticule).click();
+  });
+
+   it('validate error', () => { 
+     cy.get(IDs.btnPublish).contains('Publish Article').click();
+     
+     cy.get(`.${IDs.error}`).contains(VAR.articuleTitleInvlid)
+     cy.get(`.${IDs.error}`).contains(VAR.articuleBodyInvalid)
+     cy.get(`.${IDs.error}`).contains(VAR.articuleDescripInvalid);
```
Comment (eledwinn) [inline cypress/integration/newArticule.spec.js]: Why not configure the complete selector in the `error` variable?

File: cypress/integration/newArticule.spec.js (hunk 5 above)
Comment (eledwinn) [inline cypress/integration/newArticule.spec.js]: cammelCase please, and white spaces after comma 😺

File: cypress/integration/newArticule.spec.js [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+    });
+
+    it('Public new articule', () => { 
+
+      HAVETEXT(IDs.title,VAR.title);
+      HAVETEXT(IDs.descripcion,VAR.descripcion);
+      HAVETEXT(IDs.body,VAR.body)
+
+      cy.get(IDs.btnPublish).contains('Publish Article').click();
+      cy.wait(500);
```
Comment (eledwinn) [inline cypress/integration/newArticule.spec.js]: This wait is really necessary.

File: cypress/page-objects/loginAngular.js [hunk 10]
```diff
@@ -0,0 +1,10 @@
+
+export const IDs = {
+  idSingIn: 'a[href*="/login"]',
+  email: 'input[formcontrolname*="email"]',
+  password: 'input[formcontrolname*="password"]',
+  btnSingIn: 'button[type*="submit"]',
+  errorRegister: 'error-messages',
+  nameProfile: 'a[href*="/profile/prueba%20name"]',
+
```
Comment (eledwinn) [inline cypress/page-objects/loginAngular.js]: Remove empty line :)

File: cypress/page-objects/newArticule.js [hunk 11]
```diff
@@ -0,0 +1,10 @@
+export const IDs = {
+
```
Comment (eledwinn) [inline cypress/page-objects/newArticule.js]: Remove empty line :), and use simple quotes.

**Review by eledwinn — CHANGES_REQUESTED** [review]

File: cypress/integration/newArticule.spec.js (hunk 9 above)
Comment (Erikaguilera123) [reply cypress/integration/newArticule.spec.js]: yes, because  return error the test

File: cypress/const/constants.js [hunk 12]
```diff
@@ -1,15 +1,23 @@
...
-  passInvalid:'password is too short (minimum is 8 characters)',
-  userInvalid:"username can't be blank"
+  passInvalid: 'password is too short (minimum is 8 characters)',
+  userInvalid: "username can't be blank",
+  emailRegister: 'correoprueba2@correo.co',
+  passRegister: '1234567890',
+  error: 'email or password is invalid',
+  articuleTitleInvlid: "title can't be blank,is too short (minimum is 1 character)",
+  articuleBodyInvalid: "body can't be blank",
+  articuleDescripInvalid: "description can't be blank,is too short (minimum is 1 character)",
```
Comment (eledwinn) [inline cypress/const/constants.js:15]: Single quotes please

**Review by eledwinn — CHANGES_REQUESTED** [review]

**Review by eledwinn — APPROVED** [review]

File: cypress/const/constants.js (hunk 12 above)
Comment (Erikaguilera123) [reply cypress/const/constants.js:15]: double quotes are required

Comment (betoobernal) [reply cypress/const/constants.js:15]: escape quotes `\'` of the contractions

**Review by betoobernal — APPROVED** [review]

**Pull request merged**